src/metabase/driver.clj

```diff
@@ -967,7 +967,7 @@
 (defmulti alter-columns!
   "Alter columns given by `column-definitions` to a table named `table-name`. If the table doesn't exist it will throw an error.
   Currently we do not currently support changing the the primary key, or take any guidance on how to coerce values."
-  {:added "0.49.0", :arglists '([driver db-id table-name column-definitions])}
+  {:added "0.49.0", :arglists '([driver db-id table-name column-definitions & args])}
```

Comment:
Technically this is a breaking change to the interface of `alter-columns!`. If you're going to do this you need to make a driver changelog entry. Perhaps this is YAGNI

Reply:
The asymmetry between the methods bothers me slightly, as does the prospect of a future breaking change - better to break thing before wide adoption.

Will YAGNI for now to get this merged, and hammock a bit on whether we'll ever actually need settings.

Reply:
To be clear YAGNI for me means removing the extra args.

I don't mind either way, I am also bothered by the assymetry. But if you keep it, add it to the changelog.

Reply:
Aligned. Curious what else we could have meant by YAGNI 😄